### PR TITLE
Unlimited (100x) undo buffer with compression

### PR DESCRIPTION
**Pull request by pascal-niklaus**
This (small) PR implements LZ4 compression of the undo buffers.

LZ4 compression/decompression is very fast, and with the drawings I tested, achieves a compression down to 1.5-2.5% of the original buffer. Hence, this allows for a very deep undo buffer without significant memory penalty. 
I have fixed it at 100 which for practical purposes is unlimited.

The entire logic of the undo system remains unaltered. 

The only difference is that `swap_surface` includes a compression/decompression step.

The undo buffers are only allocated once they are needed, and they are successively grown when this is necessary.

### CODE REVIEW

File: src/config.c [hunk 1]
```diff
@@ -316,6 +316,7 @@ gboolean parse_config (GromitData *data)
                   g_printerr ("WARNING: Unable to copy \"%s\": "
                               "not yet defined!\n", copy);
                 }
+              g_free(copy);
```
Comment (pascal-niklaus) [inline src/config.c:319]: Sorry, just realized I included this here. It fixes a memory leak.

**Comment by pascal-niklaus** [comment]
Sorry, I have parts in the PR that were not meant to be in there. Not sure how this happens. I will tidy this up now...

File: src/main.c [hunk 2]
```diff
@@ -424,15 +428,14 @@ void select_tool (GromitData *data,
...
 
 
-
 void snap_undo_state (GromitData *data)
 {
   if(data->debug)
     g_printerr ("DEBUG: Snapping undo buffer %d.\n", data->undo_head);
 
-  copy_surface(data->undobuffer[data->undo_head], data->backbuffer);
+  undo_compress(data, data->backbuffer);
```
Comment (bk138) [inline src/main.c:433]: Why not have compression and saving in one function?

File: src/main.h [hunk 3]
```diff
@@ -193,4 +201,5 @@ void paint_context_free (GromitPaintContext *context);
 
 void indicate_active(GromitData *data, gboolean YESNO);
```
Comment (bk138) [inline src/main.h:203]: Seems unused.

**Review by bk138 — COMMENTED** [review]
Very quick first review while commuting, probably more to come 😉

**Pull request merged**